[FIX] sale_order_type_ux: When creating downpayment product unset company_id to avoid multi-company issues

rov-adhoc opened the pull request:
Unset the 'company_id' on the product created for down payments 
by overriding '_prepare_down_payment_product_values'.

This avoids issues when creating invoices in a different company 
than the one that originally created the down payment product, especially in multi-company environments.

rov-adhoc committed:
[FIX] sale_order_type_ux: When creating downpayment product unset company_id to avoid multi-company issues

Unset the 'company_id' on the product created for down payments by overriding
'_prepare_down_payment_product_values'.

This avoids issues when creating invoices in a different company than the one
that originally created the down payment product, especially in multi-company
environments.

roboadhoc commented:
[Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/sale/pull/1183).

rov-adhoc committed:
[FIX] sale_order_type_ux: Disable check_company on tax_id to allow multi-company advance invoices

In certain cases, a sales order may belong to one company while its related advance invoice
is created in another. When this happens, assigning taxes to the invoice would raise a
ValidationError due to mismatched company IDs on the Many2many tax_id field.

To prevent this, we explicitly disable check_company on the tax_id field. This allows taxes
from a different company to be assigned when necessary, particularly in multi-company flows
involving down payments or prepayments.

This change is intentional and limited in scope. Company consistency should still be ensured
at the business logic level where appropriate.

cav-adhoc commented:
@roboadhoc r+

roboadhoc commented:
@rov-adhoc @cav-adhoc because this PR has multiple commits, I need to know how to merge it:

* `merge` to merge directly, using the PR as merge commit message
* `rebase-merge` to rebase and merge, using the PR as merge commit message
* `rebase-ff` to rebase and fast-forward

cav-adhoc commented:
@roboadhoc rebase-ff

roboadhoc commented:
Merge method set to rebase and fast-forward.
